packages/react/src/types.ts

```diff
@@ -0,0 +1,24 @@
+type AsProp<C extends React.ElementType> = {
```

Comment:
You shouldn’t use the `React` namespace, as this leads to TypeScript generating invalid type definitions. Instead, use named imports.